README.md

```diff
@@ -47,4 +47,8 @@ Data that satisfies the non-cryptographic part of the Request.
 
 ### ZKL Flow
 
+Holders don't handle cryptographic material or messaging with Verifiers directly, but instead use software and hardware components to do this called agents.
+Cryptographic secrets are stored in wallets. Crypto engine is the software/hardware library that performs the ZK computations. Public registry
+is any reliable source that stores the public material that both Holder and Verifier. Sovrin is a public permissioned ledger designed for this purpose.
```

Comment:
Expand on "reliable", so "public registry" is a highly available and tamper-evident storage.